chore: [ADR-0004] use RadixUI (#741)
asyncapi/studio: doc/adr/0004-use-radix-ui.md

# 1. Record architecture decisions

Date: 2023-07-17

## Status

Accepted

## Context

> In our project, we have identified the need for a set of UI components that handle not only the logical part but also the design aspects. We have evaluated existing libraries such as Radix and Headless UI, which provide components with implicit designs. 

Reviewer: ```suggestion
In our project, we have identified the need for a set of UI components that only handle the logical part without imposing any design aspects. We have evaluated existing libraries such as Radix and Headless UI, which provide components without implicit designs. 
```

It's actually the opposite 😄

> We have concluded that using a ready-made library would restrict our freedom to customize the look and feel of the components. Additionally, these libraries integrate design elements into their components, making it challenging or impossible to modify them according to our specific requirements.

Reviewer: ```suggestion
We have concluded that using a ready-made library would restrict our freedom to customize the look and feel of the components. Additionally, this kind of libraries integrate design elements into their components, making it challenging or impossible to modify them according to our specific requirements.
```
